src/plugin-hooks/useRowSelect.js

```diff
@@ -256,9 +256,12 @@ function useInstance(instance) {
     return selectedFlatRows
   }, [rows, selectSubRows, selectedRowIds, getSubRows])
 
-  let isAllRowsSelected = Boolean(
-    Object.keys(nonGroupedRowsById).length && Object.keys(selectedRowIds).length
-  )
+  let isAllRowsSelected = React.useMemo(() => {
```

Comment:
Reason for doing this - Object.keys(obj).length is a costly operation, when obj is having ~100K of keys, it takes around ~100-150ms time, so memoizing this can save time when these values are actually not required to be re-computed.